modules/identity-shiro/src/internalClusterTest/java/org/opensearch/identity/HttpSmokeTestCaseWithIdentity.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.identity.shiro;
+
+import com.carrotsearch.randomizedtesting.annotations.ThreadLeakScope;
+import org.opensearch.common.network.NetworkModule;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.plugins.Plugin;
+import org.opensearch.test.OpenSearchIntegTestCase;
+import org.opensearch.transport.Netty4ModulePlugin;
+import org.opensearch.transport.nio.MockNioTransportPlugin;
+import org.opensearch.transport.nio.NioTransportPlugin;
+import org.junit.BeforeClass;
+import org.opensearch.common.util.FeatureFlags;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+/**
+ * Abstract Rest Test Case  installs and enables ShiroIdentityPlugin and removes mock
+ * http transport to enable REST requests against a test cluster
+ *
+ * @opensearch.experimental
+ */
+// TODO not sure why ThreadLeakScope.NONE is required
+@ThreadLeakScope(ThreadLeakScope.Scope.NONE)
```

Comment:
We probably could remove the `@ThreadLeakScope(ThreadLeakScope.Scope.NONE)`, the thread leaks have to be detected